concepts/modularization/migration.md

```diff
@@ -0,0 +1,32 @@
+# Migration plan
+
+Migration plan will be executed several times until all modules are moved from reconciler to lifecycle-manager. Is it possible to migrate several modules at once but it is not recommended. The migration plan uses `serverless` module as an example.
+
+The prerequisite for migration is that lifecycle-manager is already activated in KCP for all the clusters (Kyma CRs are created in the KCP and SKR). Migration is split into 2 phases: enabling module for the new clusters and update existing clusters.
+
+![migration](assets/migration.drawio.svg)
+
+## Enable module for new clusters (first)
+
+First step is to enable module installation with lifecycle manager for new clusters. Existing clusters are still managed by reconciler.
+- Disable module template replication for existing clusters (moduleCatalog=false). This prevents users to add serverless module before it is disabled in the reconciler.
+- New Kyma version is enabled in KEB (without serverless module). New clusters do not have serverless until customer enabled it in Kyma resource.
+- Serverless operator(manager) is ready and module template is available in the fast/regular channel. 
+
+## Update existing clusters
+
+Migration can be stopped at any moment if any problem occurs. Not migrated clusters will still use serverless reconciler (old Kyma version with serverless)
+
+- Upgrade the clusters to the new Kyma version (without serverless) - from this point reconciler doesn't touch serverless resources.
```

Comment:
Does this step actually removes installed serverless (workloads) ? or we simply switch off reconciliation os serverless?
 
 It what point (if any ) are the "old serverless" workloads removed?
 I think once we apply Serverless CR we will have duplicated serverless workloads, no?
 
 Or is it expected that serverless operator "takes over" existing installation , i.e by annotating them

Reply:
We do not want to remove (uninstall) serverless during migration. Customers expect continuation. We switch off reconciliation and install the serverless operator. The operator has to take ownership of resources created by the reconciler. How it is done - is up to you.